xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+import subprocess
+import re
+import os
+from typing import Dict, List
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+
+
+@dataclass
+class Attribute_Tests:
+    attribute_name: str
+    attribute_path: str
+    tests: List[str]
+    output_result: Dict[str, List[str]]
+    expect_output_to_be_equal: Dict[str, bool]
+    expect_files_to_be_equal: Dict[str, bool]
+
+
+def run_xml_converter(*args):
+    try:
+        # Build the command to execute the C++ program with the desired function and arguments
+        cmd = [xml_converter] + list(args)
```

Comment:
Lets have the args be broken out in this instead of asking them to be passed in;
eg
```
cmd = [xml_converter, "--input-filez", input_file, "--output-documents", output_file]
```

Reply:
It's possible that we will want to run this command without all of the possible arguments. There are currently 5 that can be used in xml_converter. In this case, I am reading an xml and outputing xml but I plan on adding in reading waypoint files.

Reply:
then make the arguments something like:
```python
def run_xml_converter(input_xml: Optional[str], output_xml: Optional[str], input_proto: Optional[str]... etc
```
to abstract the actual functionality for calling the binary. This way you can even re-use the same function or function signature for the library-based tests in the future if that is something that we want to do.